src/equations/bbm_1d.jl

```diff
@@ -0,0 +1,196 @@
+# TODO: Use this nondimensionalized version or use dimensional version including gravity and depth?
```

Comment:
I am not sure whether we should implement the nondimensionalized version as here, which is, AFAICT, more commonly used in the literature (e.g., also in "A Broad Class of Conservative Numerical Methods for Dispersive Wave Equations") or whether we should implement a dimensional form containing the gravitational constant and the (flat or maybe in the future even variable) bathymetry. I would prefer this because it is more consistent to the other equations and is essentially more general because this version can be achieved by setting the constants to 1. A downside, however, would be to invest a little bit more time to generalize the semidiscretization and some other functions. What are your thoughts, @ranocha?

Reply:
This is a question I have been thinking about for quite some time - and I agree with you, it would be better to use the dimensional form to be able to compare result for physical setups.

Reply:
Ok, do you know of a good reference, which uses the dimensional form of the BBM equation and presents the exact coefficients in front of each term (e.g., I guess for the $\eta_{xxt}$ it would be $\frac{1}{6}D^2$)? I could probably come up with it by my own, but it would be nice to have something to refer.

Reply:
A related question: What is actually the velocity in the case of the BBM equation? I guess it is somehow (assumed to be) related to the water height by some algebraic relation instead of by a coupled PDE? If so, how does this relation (assumption) look like exactly? It could be nice to make `velocity(q, equations::BBMEquation1D)` work to return this relation given $\eta$. I think I'm missing a bit the (physical) basics/assumptions of these scalar equations.

Reply:
If we use a dimensional form, it would probably be also good to have the bathymetry part of the unknowns again (i.e. have two unknowns, $\eta$ and $D$) for consistency with the other equations and to be able to generalize it to a variable bathymetry in the future.

Reply:
I don't have a good reference for the physical setup in mind. Maybe starting with the original paper of BEnjamin, Bona, and Mahony would be a good idea? Or we could ask Dimitrios Mitsotakis maybe?

Reply:
I'll make a bit of research.

Reply:
I think you can take most from the KdV equation, e.g., https://de.wikipedia.org/wiki/Korteweg-de-Vries-Gleichung

Reply:
So this means in our notation, the KdV equation reads in dimensional variables:

$$
\eta_t + \sqrt{gD}\eta_x + 3/2\sqrt{g/D}\eta\eta_x + 1/6\sqrt{gD}D^2\eta_{xxx} = 0
$$

For the BBM equation this would mean:

$$
\eta_t + \sqrt{gD}\eta_x + 3/2\sqrt{g/D}\eta\eta_x - 1/6D^2\eta_{xxt} = 0
$$

I also looked this up in [Whitham, _Linear and Nonlinear Waves_ (1974)](https://eclass.uoa.gr/modules/document/file.php/PHYS289/%CE%92%CE%B9%CE%B2%CE%BB%CE%AF%CE%B1/Whitham_Linear_and_Nonlinear_Waves.pdf) and found these two forms of the KdV and BBM equations in equation (13.99) and unnamed equation after (13.99), where $\gamma$ is defined in (13.94), $c_0 = \sqrt{gh_0}$ and $h_0$ is the undisturbed water depth, which we call $D$. I also double-checked that a nondimensionalization with $u = \eta/D$, $\tilde x = x/D$, $\tilde t = t\sqrt{g/D}$ all coefficients cancel and for the BBM equation we get the version, which is implemented now with the additional factors $3/2$ and $1/6$. If you agree @ranocha, I would put $D$ as a second variable into the equations, put an additional parameter `gravity` in the equations, and include the coefficients into the semidiscretizations and in the initial conditions, which are implemented now.

Reply:
Nice 👍 

Why do you want to introduce $D$ as a second variable into the equations? Isn't it assumed to be constant for the equations above?

Reply:
Yes, $D$ is assumed to be constant. I wanted to include it nervertheless to be consistent with the other equations and to be able to generalize it to a space- and/or time-dependent bathmetry. But since this is probably never happening, we can also put it as a parameter into the equations. I think it's just a bit odd that in this case $D$ is set in the initial condition for the other equations and as parameter for the BBM equation. On the other hand it might be confusing to set it in the initial condition if it has to be constant.

Reply:
Yes, I agree. I don't see a good reason to look for a variable-bathymetry formulation for simple one-way wave models like these. It makes much more sense to include bidirectional waves first before thinking about variable bathymetry. Thus, I would prefer setting the constant in the equations, even if it is not perfectly consistent with the other equations.

Reply:
Ok, makes sense. Thanks for your insights!